gamification/process/update.py

```diff
@@ -135,6 +137,45 @@ def get_participant_objects():
     return participants
 
 
+def get_meta_review_participant_objects():
+    return meta_review_participants.objects.all()
+
+
+meta_review_completed_list = []
+
+
+def update_participants_data_with_meta_review(meta_review_participant):
+    """
+    Update participants based on meta-review
+
+    This method updates every participant based on the meta-review
+    received or given. If the  meta-review participant is
+    in the active newcomers list then it will check if the meta-review
+    is complete or not and update the activity accordingly.
+    """
+    active_newcomers_list = active_newcomers()
+    if (meta_review_participant.login in active_newcomers_list and
+            meta_review_participant.login not in meta_review_completed_list):
+        # Get the corresponding gamification participant
+        gamification_participant = Participant.objects.get(
+            username=meta_review_participant.login)
+        reaction = Reaction.objects.get(id=meta_review_participant.login)
```

Comment:
This is definitely wrong. I checked the output you print out https://pastebin.com/6X6y5Kfx. You have the illusion that this works, because your test is wrong. In your test, both participant and reaction share the same id (primary_key), which gives you the wrong belief that you are correct.

That's why I said you should run `.ci/build.sh` and show the log to prove this works. You can use that for debugging, but you don't need to prove anymore as I am pretty confident now this is wrong.

Reply:
First you need to fix the test case. See https://github.com/coala/community/pull/190#discussion_r229155532 so that your test won't pass (which is good coz now you know your code is wrong).

Then read the definition of `Reaction` model. Understand the relationships between those models.

Reply:
In order to test things, I used my user handle ```shikharvaish28```. Ideally, I should be able to get  ```reaction``` using ```id``` which is the primary key of ```Reaction``` model.

```python
test_meta_review_participant = meta_review_participant.objects.get(login='shikharvaish28')
reaction = Reaction.objects.get(id=test_meta_review_participant.login)
``` 
Instead, I am getting ```DoesNotExist: Reaction matching query does not exist.``` Can you please tell me why is it so?

Reply:
> Since login is the primary key of Participant, I am expecting it to be linked in someway to id of Reaction.

Your assumption is wrong. Django does not work like that.

I strongly recommend you to read https://docs.djangoproject.com/en/2.1/topics/db/queries/, esepecially understand how Foreign Key works.